### ISSUE TITLE
V55 causes some WPF graphical artifacts on some machines.

### ISSUE THREAD

**Issue by MartinStensaker** [cefsharp/CefSharp#1961]
- **What version of the product are you using?**
NuGet: CefSharp.Wpf 55.0.0

- **What architecture x86 or x64?**
AnyCpu solution based on [1714](https://github.com/cefsharp/CefSharp/issues/1714) (the machine causing issues is x64)

- **On what operating system?**
Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
WPF

- **What steps will reproduce the problem?**
Include CefSharp, apply the solution for AnyCpu, run in Win10. 

- **What is the expected output? What do you see instead?**
Other WPF-components outside of CefSharp are glitching out when CefSharp is included. Images are drawn out and blurred and other strange artifacts appear. We have tried disabling the gpu acceleration for CefSharp without success. However, setting `RenderOptions.ProcessRenderMode = System.Windows.Interop.RenderMode.SoftwareOnly;` in the application process removes the artifacts. Also, chrome://gpu/ shows no information, only the headers inside the page are shown when viewing the page, not the usual listing of hardware capabilities and such.

- **Please provide any additional information below.**
We are using Telerik components for some elements. CefSharp itself renders perfectly normal. There are no issues on 3 out of the 4 tested machines, however more testing will be done. 

- **Does this problem also occur in the `CEF` Sample Application from http://opensource.spotify.com/cefbuilds/index.html?**
No

**Comment by amaitland** [comment]
> We have tried disabling the gpu acceleration for CefSharp without success.

What command line args did you use? What does the `Debug.log` file say?

> Also, chrome://gpu/ shows no information, only the headers inside the page are shown when viewing the page, not the usual listing of hardware capabilities and such.

What `GPU` does the machine have? Are you running the latest drivers? Does the laptop have `Nvidia Optimus` or similar? If so have you forced with `GPU` is used? 

In general it would suggest an incompatibility with your `GPU`.

It's important to remember that `CefSharp` is just a wrapper, in all likelihood you'll either need to switch the `WinForms` version hosted in `WPF` or upgrade/downgrade versions to find one that's compatible with this machine (There isn't currently a newer version).

> Does this problem also occur in the CEF Sample Application from http://opensource.spotify.com/cefbuilds/index.html?

When you open `chrome://gpu/` in `cefclient` what do you get? Make sure your using the exact version and using ` cefclient --multi-threaded-message-loop --off-screen-rendering-enabled`

**Comment by amaitland** [comment]
> WPF graphical artifacts on some machines

Have you experienced this issue on more than one machine?

**Comment by MartinStensaker** [comment]
We seem to have stumbled upon a solution. When we moved the AnyCpu-fixes from the application constructor to the application startup-event `Application.Current.Startup` the issues are not showing up any longer. We are not sure why this is the case or if something automatically fixed itself on the machine during the night (Windows 10, you know)...

> What command line args did you use? What does the Debug.log file say?

We tried all kinds of `Cef.CefCommandLineArgs` (like all the `disable-gpu*` type flags). We also tried the `settings.DisableGpuAcceleration` and `settings.SetOffScreenRenderingBestPerformanceArgs`. Nothing really improved the situation on that particular machine. Debug.log is empty.

> What GPU does the machine have? Are you running the latest drivers? Does the laptop have Nvidia Optimus or similar? If so have you forced with GPU is used?

Yes, the machine is sporting an Nvidia 950M graphics card combined with Intel graphics. And I guess that could cause some of the problems as you suggested. We also tried updating all drivers without luck.

> When you open chrome://gpu/ in cefclient what do you get? Make sure your using the exact version and using cefclient --multi-threaded-message-loop --off-screen-rendering-enabled

Works as intended, no problems with this client. 

> Have you experienced this issue on more than one machine?

No, our test bench consists of all kinds of hardware (as well as several Nvidia/Intel hybrids) but this is the only one having issues.

**Comment by amaitland** [comment]
Did you revert your changes to validate your fix? Did you look at Windows update to see if new updates were installed?

AnyCpu support sounds very unlikely as the cause, anything is possible though.

Chromium has a long history of problems with those hybrid GPUs.

**Comment by MartinStensaker** [comment]
> Did you revert your changes to validate your fix? Did you look at Windows update to see if new updates were installed?

The problems reappeared when we reverted the fix, so this fix is valid. No windows updated were installed last time either. 

> AnyCpu support sounds very unlikely as the cause, anything is possible though.

Yes, I agree - and I should have been more clear as to what else was going on in the `InitializeCefSharp`-function that we moved. It also contained all the init for CefSharp, like `Cef.Initialize(settings, performDependencyCheck: false, browserProcessHandler: null)` with the proper `CefSettings` supplied. 

Not sure why moving the `InitializeCefSharp`-function from the constructor to the Application_Startup-event resolved the issue on that particular machine - everything just worked either way on the other machines. Thanks for the help @amaitland with this issue and for guiding us in the continued troubleshooting. Maybe this will help someone else troubleshooting their edge cases with similar graphical glitches on similar machines.
